hedera-node/src/main/java/com/hedera/services/store/contracts/ContractsState.java

```diff
@@ -157,7 +172,7 @@ public void commit() {
 						/* Allow creation only for contract accounts */
 						ledger.create(pair.getKey(), accountId, 0L, pair.getValue());
 					} else {
-						// TODO throw exception, account creation not allowed, resolve after refactoring
+						throw new InvalidTransactionException(ResponseCodeEnum.INVALID_SOLIDITY_ADDRESS);
 					}
```

Comment:
Instead of using the `if/else` approach, we can use the `validateFalse` functions used in TypedTokenStore and AccountStore to throw exceptions etc

Reply:
done